### ISSUE TITLE
Armour Duplication SyncNBT related??

### ISSUE THREAD

**Issue by Elad-PenDragon** [nsg/SyncNBT#13]
How to reproduce : 
Goto for example Bandomeer with armour equiped.
Remove armour but keep in inventory.
Use /home <homename> command to teleport to a home on another world(server) for example Rori.
You will enter Rori with the armour in your inventory AND equiped.

**Comment by nsg** [comment]
Sounds like I have forgotten to properly clear the inventory slots. I made the code a little more "relaxed" earlier when there was a lot of bugs to prevent to much item loss. I still have two problematic bugs #9 and #15 what I want to fix soon that probably will help.

**Comment by nsg** [comment]
This bug is most likely fixed with version 0.3, TODO: test it.

**Comment by Kainzo** [comment]
This will happen if a player connects to a new server before disconnect from the last.

If you adopt Spigot's Bungee proxy service you could send plugin messages to have the new server and old server sync before connecting the player.

**Comment by nsg** [comment]
No, not a race condition. Protection against that was added in day one. I write a lock in the database. I choose to not depend on Bungeecord if someone choose to use some other means to to transfer players.

There was a lot of accidental inventory wipes in 0.2 from bugs and other problems. We are still pre-1.0 so it is still early testing. I actually relaxed some part of the code so it was less likely to loose items (to be nice to my beta-testing players), with the side effect with a few dupes. This bug do not apply in 0.3, I will close this ticket.
